### PR TITLE
[Aio] Extend unit tests for async credentials calls

### PR DESCRIPTION
**Pull request by lidizheng**
Forked from https://github.com/grpc/grpc/pull/21920, original author @rmariano.

This PR meant to make our CI happy with the change.

---

Fixes https://github.com/grpc/grpc/issues/22302

### CODE REVIEW

**Comment by lidizheng** [comment]
Somehow this bug is not reproducible locally. It could be a bug in 3.6, or an environmental difference in Kokoro machines.

**Comment by lidizheng** [comment]
On a slow CPU, no matter 3.7 or 3.6, we can reproduce the error:
```
Traceback (most recent call last):
  File "/usr/lib/python3.7/asyncio/base_events.py", line 400, in create_task
    task = tasks.Task(coro, loop=self)
  File "/usr/lib/python3.7/asyncio/base_events.py", line 685, in call_soon
    self._check_thread()
  File "/usr/lib/python3.7/asyncio/base_events.py", line 723, in _check_thread
    "Non-thread-safe operation invoked on an event loop other "
RuntimeError: Non-thread-safe operation invoked on an event loop other than the current one
Exception ignored in: 'grpc._cython.cygrpc._AsyncioSocket.write'
Traceback (most recent call last):
  File "/usr/lib/python3.7/asyncio/base_events.py", line 400, in create_task
    task = tasks.Task(coro, loop=self)
  File "/usr/lib/python3.7/asyncio/base_events.py", line 685, in call_soon
    self._check_thread()
  File "/usr/lib/python3.7/asyncio/base_events.py", line 723, in _check_thread
    "Non-thread-safe operation invoked on an event loop other "
RuntimeError: Non-thread-safe operation invoked on an event loop other than the current one
```

**Comment by lidizheng** [comment]
Interestingly, the metadata authenticate plugin calls a Python function in a new thread by default. The new thread will return the metadata hence clears the block for request sending. Core then invokes `write` on socket, which resulted in the exception above. This bug becomes quite difficult to track down because after the exception was raised, the newly spawned thread died out left no trace.

The fix is simple that instead of spawning a new thread for the callback, we can schedule the callback onto the event loop.

**Comment by lidizheng** [comment]
Unfortunately, since the Async API is not the default stack, classes like `Credentials` will call `init_grpc()` which crashes custom IO manager. So, users still have to call `init_grpc_aio()` explicitly if they want to use the custom IO manager.

**Comment by lidizheng** [comment]
@gnossen  @rmariano PTAL.
CC @pfreixes

File: src/python/grpcio/grpc/_cython/_cygrpc/aio/grpc_aio.pyx.pxi [hunk 1]
```diff
@@ -114,7 +114,7 @@ cdef _actual_aio_shutdown():
         raise ValueError('Unsupported engine type [%s]' % _global_aio_state.engine)
 
 
-cdef init_grpc_aio():
+cpdef init_grpc_aio():
     """Initialis the gRPC AsyncIO module.
```
Comment (gnossen) [inline src/python/grpcio/grpc/_cython/_cygrpc/aio/grpc_aio.pyx.pxi]: Nit: I realize this isn't actually part of this PR, but I just now noticed: s/Initialis/Initializes/

Comment (lidizheng) [reply src/python/grpcio/grpc/_cython/_cygrpc/aio/grpc_aio.pyx.pxi]: Good catch 🤣

**Comment by lidizheng** [comment]
Known failures: https://github.com/grpc/grpc/issues/22351

**Pull request merged**